{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Moving window analysis (MWA) of cyano BGC\n",
    "\n",
    "This document outlines the workflow implemented for a moving window analysis of specific modules of cyanobacterial natural product producing BGCs (biosynthetic gene clusters). "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Creation of software environment to do the analysis\n",
    "\n",
    "This is the prologue to create a bioinformatics software environment for the required moving window analysis. \n",
    "\n",
    "Note: This needs to be done only once and need not be repeated everytime the workflow is carried out. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Initialise the conda environment if not already initialised\n",
    "source activate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "r-cli-2.0.2          | 396 KB    | ##################################### | 100% \n",
      "r-plogr-0.2.0        | 19 KB     | ##################################### | 100% \n",
      "r-optparse-1.6.6     | 79 KB     | ##################################### | 100% \n",
      "r-generics-0.0.2     | 75 KB     | ##################################### | 100% \n",
      "r-purrr-0.3.4        | 407 KB    | ##################################### | 100% \n",
      "r-rprojroot-1.3_2    | 94 KB     | ##################################### | 100% \n",
      "r-whisker-0.4        | 80 KB     | ##################################### | 100% \n",
      "r-lattice-0.20_41    | 1.1 MB    | ##################################### | 100% \n",
      "freetype-2.10.2      | 874 KB    | ##################################### | 100% \n",
      "r-backports-1.1.7    | 80 KB     | ##################################### | 100% \n",
      "r-fs-1.4.1           | 290 KB    | ##################################### | 100% \n",
      "r-processx-3.4.2     | 257 KB    | ##################################### | 100% \n",
      "r-selectr-0.4_2      | 449 KB    | ##################################### | 100% \n",
      "r-assertthat-0.2.1   | 70 KB     | ##################################### | 100% \n",
      "r-dbi-1.1.0          | 669 KB    | ##################################### | 100% \n",
      "r-dplyr-0.8.5        | 1.9 MB    | ##################################### | 100% \n",
      "r-forcats-0.5.0      | 375 KB    | ##################################### | 100% \n",
      "r-readxl-1.3.1       | 855 KB    | ##################################### | 100% \n",
      "r-scales-1.1.1       | 556 KB    | ##################################### | 100% \n",
      "setuptools-46.4.0    | 653 KB    | ##################################### | 100% \n",
      "r-tidyselect-1.1.0   | 204 KB    | ##################################### | 100% \n",
      "r-tibble-3.0.1       | 378 KB    | ##################################### | 100% \n",
      "r-tidyverse-1.3.0    | 452 KB    | ##################################### | 100% \n",
      "r-rmarkdown-2.1      | 3.0 MB    | ##################################### | 100% \n",
      "r-ggplot2-3.3.0      | 3.9 MB    | ##################################### | 100% \n",
      "r-rlang-0.4.6        | 1.0 MB    | ##################################### | 100% \n",
      "r-reshape2-1.4.4     | 126 KB    | ##################################### | 100% \n",
      "r-pillar-1.4.4       | 197 KB    | ##################################### | 100% \n",
      "r-rvest-0.3.5        | 656 KB    | ##################################### | 100% \n",
      "r-yaml-2.2.1         | 111 KB    | ##################################### | 100% \n",
      "r-markdown-1.1       | 140 KB    | ##################################### | 100% \n",
      "r-base-4.0.0         | 23.3 MB   | ##################################### | 100% \n",
      "r-ps-1.3.3           | 224 KB    | ##################################### | 100% \n",
      "r-askpass-1.1        | 27 KB     | ##################################### | 100% \n",
      "r-base64enc-0.1_3    | 42 KB     | ##################################### | 100% \n",
      "r-xfun-0.14          | 205 KB    | ##################################### | 100% \n",
      "r-xml2-1.3.2         | 247 KB    | ##################################### | 100% \n",
      "r-munsell-0.5.0      | 246 KB    | ##################################### | 100% \n",
      "r-pkgbuild-1.0.8     | 153 KB    | ##################################### | 100% \n",
      "r-gtable-0.3.0       | 423 KB    | ##################################### | 100% \n",
      "r-curl-4.3           | 697 KB    | ##################################### | 100% \n",
      "pip-20.1.1           | 1.1 MB    | ##################################### | 100% \n",
      "r-rematch-1.0.1      | 19 KB     | ##################################### | 100% \n",
      "r-utf8-1.1.4         | 155 KB    | ##################################### | 100% \n",
      "r-httr-1.4.1         | 490 KB    | ##################################### | 100% \n",
      "r-r6-2.4.1           | 63 KB     | ##################################### | 100% \n",
      "r-mime-0.9           | 50 KB     | ##################################### | 100% \n",
      "r-sys-3.3            | 46 KB     | ##################################### | 100% \n",
      "r-rcpp-1.0.4.6       | 1.9 MB    | ##################################### | 100% \n",
      "r-mass-7.3_51.6      | 1.1 MB    | ##################################### | 100% \n",
      "r-lubridate-1.7.8    | 1.0 MB    | ##################################### | 100% \n",
      "r-htmltools-0.4.0    | 214 KB    | ##################################### | 100% \n",
      "r-cellranger-1.1.0   | 108 KB    | ##################################### | 100% \n",
      "r-jsonlite-1.6.1     | 971 KB    | ##################################### | 100% \n",
      "r-magrittr-1.5       | 167 KB    | ##################################### | 100% \n",
      "r-mgcv-1.8_31        | 2.7 MB    | ##################################### | 100% \n",
      "r-prettyunits-1.1.1  | 40 KB     | ##################################### | 100% \n",
      "r-praise-1.0.0       | 23 KB     | ##################################### | 100% \n",
      "r-plyr-1.8.6         | 831 KB    | ##################################### | 100% \n",
      "r-tinytex-0.23       | 113 KB    | ##################################### | 100% \n",
      "clangxx_osx-64-9.0.1 | 14 KB     | ##################################### | 100% \n",
      "r-colorspace-1.4_1   | 2.5 MB    | ##################################### | 100% \n",
      "r-isoband-0.2.1      | 2.5 MB    | ##################################### | 100% \n",
      "r-nlme-3.1_147       | 2.3 MB    | ##################################### | 100% \n",
      "r-stringi-1.4.6      | 758 KB    | ##################################### | 100% \n",
      "r-pkgconfig-2.0.3    | 24 KB     | ##################################### | 100% \n",
      "glib-2.64.2          | 3.2 MB    | ##################################### | 100% \n",
      "r-rstudioapi-0.11    | 266 KB    | ##################################### | 100% \n",
      "r-hms-0.5.3          | 121 KB    | ##################################### | 100% \n",
      "r-labeling-0.3       | 66 KB     | ##################################### | 100% \n",
      "r-knitr-1.28         | 1.3 MB    | ##################################### | 100% \n",
      "r-openssl-1.4.1      | 1.2 MB    | ##################################### | 100% \n",
      "pcre2-10.34          | 667 KB    | ##################################### | 100% \n",
      "r-fansi-0.4.1        | 185 KB    | ##################################### | 100% \n",
      "r-matrix-1.2_18      | 3.9 MB    | ##################################### | 100% \n",
      "r-stringr-1.4.0      | 209 KB    | ##################################### | 100% \n",
      "r-pkgload-1.0.2      | 159 KB    | ##################################### | 100% \n",
      "r-testthat-2.3.2     | 1.1 MB    | ##################################### | 100% \n",
      "r-clipr-0.7.0        | 64 KB     | ##################################### | 100% \n",
      "r-reprex-0.3.0       | 429 KB    | ##################################### | 100% \n",
      "r-digest-0.6.25      | 191 KB    | ##################################### | 100% \n",
      "r-withr-2.2.0        | 227 KB    | ##################################### | 100% \n",
      "r-crayon-1.3.4       | 747 KB    | ##################################### | 100% \n",
      "r-viridislite-0.3.0  | 64 KB     | ##################################### | 100% \n",
      "r-rcolorbrewer-1.1_2 | 59 KB     | ##################################### | 100% \n",
      "r-glue-1.4.1         | 139 KB    | ##################################### | 100% \n",
      "r-farver-2.0.3       | 1.4 MB    | ##################################### | 100% \n",
      "r-broom-0.5.6        | 1.9 MB    | ##################################### | 100% \n",
      "r-progress-1.2.2     | 90 KB     | ##################################### | 100% \n",
      "r-getopt-1.20.3      | 45 KB     | ##################################### | 100% \n",
      "r-ellipsis-0.3.1     | 47 KB     | ##################################### | 100% \n",
      "r-readr-1.3.1        | 780 KB    | ##################################### | 100% \n",
      "r-vctrs-0.3.0        | 955 KB    | ##################################### | 100% \n",
      "r-evaluate-0.14      | 81 KB     | ##################################### | 100% \n",
      "r-desc-1.2.0         | 291 KB    | ##################################### | 100% \n",
      "r-highr-0.8          | 58 KB     | ##################################### | 100% \n",
      "libssh2-1.9.0        | 274 KB    | ##################################### | 100% \n",
      "r-zeallot-0.1.0      | 62 KB     | ##################################### | 100% \n",
      "r-callr-3.4.3        | 392 KB    | ##################################### | 100% \n",
      "r-dbplyr-1.4.3       | 632 KB    | ##################################### | 100% \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r-haven-2.2.0        | 344 KB    | ##################################### | 100% \n",
      "r-modelr-0.1.8       | 218 KB    | ##################################### | 100% \n",
      "r-tidyr-1.0.3        | 742 KB    | ##################################### | 100% \n",
      "r-lifecycle-0.2.0    | 112 KB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "#\n",
      "# To activate this environment, use\n",
      "#\n",
      "#     $ conda activate MWA_cyano_env\n",
      "#\n",
      "# To deactivate an active environment, use\n",
      "#\n",
      "#     $ conda deactivate\n",
      "\n",
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "conda env create -f 00-conda_recipe/MWA_cyano_env.yml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) (MWA_cyano_env) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Activate the environment\n",
    "conda activate MWA_cyano_env"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(MWA_cyano_env) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "conda list > 00-conda_recipe/env_package_list.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Start analysis for pairs\n",
    "\n",
    "We will use the package pyfasta to create custom windows where window size will be set at `50` and overlap of `25` is specified. The help documentation provides the usage of the tool."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### The bash script that runs below does a number of steps:\n",
    "- creates output folders\n",
    "- runs alignment of input files\n",
    "- replaces hyphens in the alignment with X \n",
    "- demultiplexes the file to generate individual fasta\n",
    "- fragments the demultiplexed fasta into 50aa size with 25aa overlap."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "creating new files:\n",
      "fragmented_pairs/p1/M1_nostopeptinB.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p1/M1_nostophycin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p10/M6_micropeptin88A.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p10/M7_cyanopeptolin963A.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p11/M3_oscillapeptinG.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p11/M6_oscillapeptinG.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p12/M5_oscillapeptinE.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p12/M6_oscillapeptinG.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p13/M7_cyanopeptolin963A.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p13/M8_oscillapeptinG.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p14/M2_oscillagininB.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p14/M7_cyanopeptolin963A.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p15/M2_anabaenolysinA.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p15/M2_anabaenolysinC.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p16/M2_anabaenolysinC.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p16/M6_hassallidin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p17/M2_microginin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p17/M2_oscillaginin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p18/M2_oscillaginin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p18/M7_cyanopeptolin963A.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p19/M1_nostopeptolide.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p19/M2_speudospumigin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p2/M1_nostopeptinB.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p2/M1_nostopeptinE.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p20/M2_speudospumigin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p20/M2_spumigin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p21/M3_microginin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p21/M3_oscillaginin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p22/M3_microginin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p22/M3_oscillaginin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p23/M4_bacillomycinD.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p23/M4_bacillomycinF.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p24/M4_bacillomycinD.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p24/M7_fengycin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p25/M4_microcystinLR.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p25/M4_microcystinRR.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p26/M4_microcystinRR.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p26/M6_microcystinRR.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p27/M4_anabaenopeptin915.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p27/M5_anabaenopeptin915.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p28/M5_anabaenopeptin915.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p28/M5_anabaenopeptinA.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p29/M1_nostopeptolide.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p29/M6_anabaenopeptin915.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p3/M5_micropeptinK139.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p3/M6_micropeptin88A.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p30/M6_anabaenopeptin915.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p30/M6_anabaenopeptinA.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p31/M6_mojavensin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p31/M6_mycosubtilin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p32/M6_mojavensin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p32/M7_mojavensin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p33/M1_fusaricidin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p33/M7_bacillomycinF.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p34/M7_bacillomycinF.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p34/M7_iturin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p35/M6_mycosubtilin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p35/M7_iturin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p36/M7_iturin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p36/M7_mojavensin.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p37/M2_polymyxinP.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p37/M7_polymyxinP.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p38/M7_polymyxinB.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p38/M7_polymyxinP.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p4/M5_micropeptinK139.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p4/M7_micropeptin139.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p5/M5_nostopeptinB.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p5/M5_nostopeptinE.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p6/M5_nostopeptinB.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p6/M7_nostopeptinB.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p7/M5_cyanopeptolin984.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p7/M5_oscillapeptinE.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p8/M5_oscillapeptinE.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p8/M7_cyanopeptolin984.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p9/M5_cyanopeptolin963A.split.50mer.25overlap.fa\n",
      "creating new files:\n",
      "fragmented_pairs/p9/M6_micropeptin88A.split.50mer.25overlap.fa\n",
      "(MWA_cyano_env) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bash 02-bash_scripts/gen_blast_input.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Running blast for the pairs\n",
    "\n",
    "Generate blast inputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) Warning: [blastp] Query_38 M7_cyanopeptolin963A_926: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_39 M7_cyanopeptolin963A_951: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_42 M7_cyanopeptolin963A_1026: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_43 M7_cyanopeptolin963A_1051: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_44 M7_cyanopeptolin963A_1076: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_45 M7_cyanopeptolin963A_1101: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_46 M7_cyanopeptolin963A_1126: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_47 M7_cyanopeptolin963A_1151: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_48 M7_cyanopeptolin963A_1176: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_49 M7_cyanopeptolin963A_1201: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_50 M7_cyanopeptolin963A_1226: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_51 M7_cyanopeptolin963A_1251: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_52 M7_cyanopeptolin963A_1276: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_53 M7_cyanopeptolin963A_1301: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "(MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) Warning: [blastp] Query_45 M6_hassallidin_1101: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "(MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) Warning: [blastp] Query_38 M7_cyanopeptolin963A_926: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_39 M7_cyanopeptolin963A_951: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_42 M7_cyanopeptolin963A_1026: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_43 M7_cyanopeptolin963A_1051: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_44 M7_cyanopeptolin963A_1076: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_45 M7_cyanopeptolin963A_1101: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_46 M7_cyanopeptolin963A_1126: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_47 M7_cyanopeptolin963A_1151: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_48 M7_cyanopeptolin963A_1176: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_49 M7_cyanopeptolin963A_1201: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_50 M7_cyanopeptolin963A_1226: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_51 M7_cyanopeptolin963A_1251: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_52 M7_cyanopeptolin963A_1276: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_53 M7_cyanopeptolin963A_1301: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "(MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) Warning: [blastp] Query_42 M7_mojavensin_1026: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: [blastp] Query_43 M7_mojavensin_1051: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_44 M7_mojavensin_1076: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_47 M7_mojavensin_1151: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_48 M7_mojavensin_1176: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_52 M7_mojavensin_1276: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_53 M7_mojavensin_1301: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_54 M7_mojavensin_1326: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_55 M7_mojavensin_1351: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_56 M7_mojavensin_1376: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_57 M7_mojavensin_1401: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_58 M7_mojavensin_1426: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_59 M7_mojavensin_1451: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_60 M7_mojavensin_1476: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_61 M7_mojavensin_1501: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "(MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) Warning: [blastp] Query_41 M7_iturin_1001: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_42 M7_iturin_1026: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_43 M7_iturin_1051: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "(MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) Warning: [blastp] Query_43 M7_iturin_1051: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_44 M7_iturin_1076: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_48 M7_iturin_1176: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_52 M7_iturin_1276: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_53 M7_iturin_1301: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_57 M7_iturin_1401: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_58 M7_iturin_1426: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_59 M7_iturin_1451: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_60 M7_iturin_1476: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "Warning: [blastp] Query_61 M7_iturin_1501: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options \n",
      "(MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) (MWA_cyano_env) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Pair1 M1_nostopeptinB v M1_nostophycin\n",
    "blastp -subject fragmented_pairs/p1/M1_nostopeptinB.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p1/M1_nostophycin.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair1_blast_out.txt -outfmt 6 \n",
    "\n",
    "#Pair 2 M1_nostopeptinB v M1_nostopeptinE\n",
    "blastp -subject fragmented_pairs/p2/M1_nostopeptinB.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p2/M1_nostopeptinE.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair2_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair 3 M5_micropeptinK139 v M6_micropeptin88A\n",
    "blastp -subject fragmented_pairs/p3/M5_micropeptinK139.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p3/M6_micropeptin88A.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair3_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair 4 M5_micropeptinK139 v M6_micropeptin88A\n",
    "blastp -subject fragmented_pairs/p4/M5_micropeptinK139.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p4/M7_micropeptin139.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair4_blast_out.txt -outfmt 6\n",
    "\n",
    "#Pair 5 M5_nostopeptinB v M5_nostopeptinE\n",
    "blastp -subject fragmented_pairs/p5/M5_nostopeptinB.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p5/M5_nostopeptinE.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair5_blast_out.txt -outfmt 6\n",
    "\n",
    "#Pair 6 M5_nostopeptinB v M7_nostopeptinB\n",
    "blastp -subject fragmented_pairs/p6/M5_nostopeptinB.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p6/M7_nostopeptinB.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair6_blast_out.txt -outfmt 6\n",
    "\n",
    "#Pair 7 M5_cyanopeptolin984 v M5_oscillapeptinE\n",
    "blastp -subject fragmented_pairs/p7/M5_cyanopeptolin984.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p7/M5_oscillapeptinE.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair7_blast_out.txt -outfmt 6\n",
    "\n",
    "#Pair 8 M5_oscillapeptinE v M7_cyanopeptolin984\n",
    "blastp -subject fragmented_pairs/p8/M5_oscillapeptinE.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p8/M7_cyanopeptolin984.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair8_blast_out.txt -outfmt 6\n",
    "\n",
    "#Pair 9 M5_cyanopeptolin963A v M6_micropeptin88A\n",
    "blastp -subject fragmented_pairs/p9/M5_cyanopeptolin963A.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p9/M6_micropeptin88A.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair9_blast_out.txt -outfmt 6\n",
    "\n",
    "#Pair 10 M6_micropeptin88A v M7_cyanopeptolin963A\n",
    "blastp -subject fragmented_pairs/p10/M6_micropeptin88A.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p10/M7_cyanopeptolin963A.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair10_blast_out.txt -outfmt 6\n",
    "\n",
    "#Pair 11 M3_oscillapeptinG v M6_oscillapeptinG\n",
    "blastp -subject fragmented_pairs/p11/M3_oscillapeptinG.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p11/M6_oscillapeptinG.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair11_blast_out.txt -outfmt 6\n",
    "\n",
    "#Pair 12 M5_oscillapeptinE v M6_oscillapeptinG\n",
    "blastp -subject fragmented_pairs/p12/M5_oscillapeptinE.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p12/M6_oscillapeptinG.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair12_blast_out.txt -outfmt 6\n",
    "\n",
    "#Pair 13 M7_cyanopeptolin963A v M8_oscillapeptinG\n",
    "blastp -subject fragmented_pairs/p13/M7_cyanopeptolin963A.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p13/M8_oscillapeptinG.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair13_blast_out.txt -outfmt 6\n",
    "\n",
    "#Pair 14 M2_oscillagininB v M7_cyanopeptolin963A\n",
    "blastp -subject fragmented_pairs/p14/M2_oscillagininB.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p14/M7_cyanopeptolin963A.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair14_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair15 -- M2_anabaenolysinA v M2_anabaenolysinC\n",
    "\n",
    "blastp -subject fragmented_pairs/p15/M2_anabaenolysinA.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p15/M2_anabaenolysinC.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair15_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair17 -- M2_microginin v M6_hassallidin\n",
    "blastp -subject fragmented_pairs/p17/M2_microginin.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p17/M2_oscillaginin.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair17_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair16 -- M2_anabaenolysinC v M6_hassallidin\n",
    "blastp -subject fragmented_pairs/p16/M2_anabaenolysinC.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p16/M6_hassallidin.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair16_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair18 -- M2_oscillaginin v M7_cyanopeptolin963A\n",
    "blastp -subject fragmented_pairs/p18/M2_oscillaginin.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p18/M7_cyanopeptolin963A.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair18_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair19 -- M1_nostopeptolide v M2_speudospumigin\n",
    "blastp -subject fragmented_pairs/p19/M1_nostopeptolide.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p19/M2_speudospumigin.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair19_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair20 -- M2_speudospumigin v M2_spumigin\n",
    "blastp -subject fragmented_pairs/p20/M2_speudospumigin.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p20/M2_spumigin.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair20_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair21 -- M3_microginin v M3_oscillaginin\n",
    "blastp -subject fragmented_pairs/p21/M3_microginin.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p21/M3_oscillaginin.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair21_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair22 -- M3_microginin v M3_oscillaginin -- Mislabeled File\n",
    "blastp -subject fragmented_pairs/p22/M3_microginin.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p22/M3_oscillaginin.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair22_mislabeledFasta_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair23 -- M4_bacillomycinD v M4_bacillomycinF \n",
    "blastp -subject fragmented_pairs/p23/M4_bacillomycinD.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p23/M4_bacillomycinF.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair23_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair24 -- M4_bacillomycinD v M7_fengycin \n",
    "blastp -subject fragmented_pairs/p24/M4_bacillomycinD.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p24/M7_fengycin.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair24_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair25 -- M4_microcystinLR v M4_microcystinRR \n",
    "blastp -subject fragmented_pairs/p25/M4_microcystinLR.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p25/M4_microcystinRR.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair25_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair26 -- M4_microcystinRR v M6_microcystinRR \n",
    "blastp -subject fragmented_pairs/p26/M4_microcystinRR.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p26/M6_microcystinRR.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair26_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair27 -- M4_anabaenopeptin915 v M5_anabaenopeptin915 \n",
    "blastp -subject fragmented_pairs/p27/M4_anabaenopeptin915.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p27/M5_anabaenopeptin915.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair27_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair28 -- M5_anabaenopeptin915 v M5_anabaenopeptinA \n",
    "blastp -subject fragmented_pairs/p28/M5_anabaenopeptin915.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p28/M5_anabaenopeptinA.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair28_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair29 -- M1_nostopeptolide v M6_anabaenopeptin915 \n",
    "blastp -subject fragmented_pairs/p29/M1_nostopeptolide.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p29/M6_anabaenopeptin915.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair29_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair30 -- M6_anabaenopeptin915 v M6_anabaenopeptinA \n",
    "blastp -subject fragmented_pairs/p30/M6_anabaenopeptin915.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p30/M6_anabaenopeptinA.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair30_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair31 -- M6_mojavensin v M6_mycosubtilin \n",
    "blastp -subject fragmented_pairs/p31/M6_mojavensin.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p31/M6_mycosubtilin.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair31_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair32 -- M6_mojavensin v M7_mojavensin \n",
    "blastp -subject fragmented_pairs/p32/M6_mojavensin.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p32/M7_mojavensin.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair32_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair33 -- M1_fusaricidin v M7_bacillomycinF \n",
    "blastp -subject fragmented_pairs/p33/M1_fusaricidin.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p33/M7_bacillomycinF.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair33_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair34 -- M7_bacillomycinF v M7_iturin \n",
    "blastp -subject fragmented_pairs/p34/M7_bacillomycinF.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p34/M7_iturin.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair34_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair35 -- M6_mycosubtilin v M7_iturin \n",
    "blastp -subject fragmented_pairs/p35/M6_mycosubtilin.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p35/M7_iturin.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair35_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair36 -- M7_iturin v M7_mojavensin \n",
    "blastp -subject fragmented_pairs/p36/M7_iturin.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p36/M7_mojavensin.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair36_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair37 -- M2_polymyxinP v M7_polymyxinP \n",
    "blastp -subject fragmented_pairs/p37/M2_polymyxinP.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p37/M7_polymyxinP.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair37_blast_out.txt -outfmt 6\n",
    "\n",
    "# Pair38 -- M2_polymyxinP v M7_polymyxinP \n",
    "blastp -subject fragmented_pairs/p38/M7_polymyxinB.split.50mer.25overlap.fa \\\n",
    "-query fragmented_pairs/p38/M7_polymyxinP.split.50mer.25overlap.fa \\\n",
    "-out blast_out/pair38_blast_out.txt -outfmt 6\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Parse Blast Output\n",
    "\n",
    "Generates\n",
    "- Output tables with pairwise fragement blast result\n",
    "- Output pdf files of plots of fragement number and percent identity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(MWA_cyano_env) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bash 02-bash_scripts/gen_results_MWA_from_blastOut.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
